python/cuml/fil/fil.pyx

```diff
@@ -263,6 +266,7 @@ cdef class ForestInference_impl():
         self.handle = handle
         self.forest_data = forest_variant(<forest32_t> NULL)
         self.shape_str = NULL
+        self.mr = get_current_device_resource()
```

Comment:
When is `self.mr` used?

Reply:
Some Python models are storing `rmm::device_vector` and other RMM objects or structs to containing these objects. These only have a pointer to their respective memory resources. Then depending on garbage collection, the memory resource (see [memory_resource.pyx](https://github.com/rapidsai/rmm/blob/branch-22.10/python/rmm/_lib/memory_resource.pyx)) can be released before objects depending on it for CUDA deallocation. This sometimes results in a segfault (visible when benchmarking for instance). Keeping a reference to the memory resource inside the model prevents its premature release (e.g. : [in DeviceBuffer](https://github.com/rapidsai/rmm/blob/adcfb9343607bb65cfff9635ee724ba671097022/python/rmm/_lib/device_buffer.pyx#L96)).